packages/pg-pool/index.js

```diff
@@ -248,6 +251,17 @@ class Pool extends EventEmitter {
       } else {
         this.log('new client connected')
 
+        if (this.options.maxLifetimeSeconds !== 0) {
+          setTimeout(() => {
```

Comment:
Does the unit test need to be updated? The test passed.

`    ✓ unrefs the connections and timeouts so the program can exit when idle when the allowExitOnIdle option is set (151ms)`

Reply:
Actually I think an additional test is needed since `allowExitOnIdle` should be tested under both conditions (`maxLifetimeSeconds` is disabled and `maxLifetimeSeconds` is a longer time period than `idleTimeoutMillis`.)

Reply:
Sorry to also comment on this, but this is preventing a clean shutdown even when `client.end()` is called.
Can't we just use `.unref()` on the timeout or 
```js
const timeout = setTimeout(....)
client.connection.once('end', () => clearTimeout(timeout))
```

Maybe even both

Reply:
It's probably worth noting that the default setting for maxLifetimeSeconds is 0 (aka: disabled) and none of these issues appear unless you set it to a non-zero value. That doesn't resolve the issues, but it is important to be aware of since there is a simple workaround (set it back to 0.) It seems like the need to utilize maxLifetimeSeconds would be minimal in cases where clients are being explicitly disconnected (the feature was added primarily to support long-running services in load-balanced DB environments.)

Reply:
Exactly my use case (for faster recovery of a failover to a replica) 😊
I just ran into some issue with some local scripts, we're I'm using the same config. I might do a PR to fix it in the next few days, unless anybody else is on it. But haven't seen any PRs.